src/sourceTracking.ts

```diff
@@ -106,6 +107,54 @@ export class SourceTracking extends AsyncCreatable {
     this.subscribeSDREvents = options.subscribeSDREvents ?? false;
   }
 
+  /**
+   * A static method to help delete custom labels from a file, or the entire file if there are no more labels
+   *
+   * @param filename - a path to a custom labels file
+   * @param customLabels - an array of SourceComponents representing the custom labels to delete
+   */
+  public static async deleteCustomLabels(filename: string, customLabels: SourceComponent[]): Promise<void> {
+    // for custom labels, we need to remove the individual label from the xml file
+    // so we'll parse the xml
+    const parser = new XMLParser({
+      ignoreDeclaration: false,
+      ignoreAttributes: false,
+      attributeNamePrefix: '@_',
+    });
+    const cls = parser.parse(fs.readFileSync(filename, 'utf8')) as {
+      CustomLabels: { labels: Array<{ fullName: string }> | { fullName: string } };
+    };
+    if ('fullName' in cls.CustomLabels.labels) {
+      // a single custom label remains, delete the entire file
+      return fs.promises.unlink(filename);
+    } else {
+      // in theory, we should only have custom labels passed in, but filter just to make sure
+      const customLabelsToDelete = customLabels
+        .filter((label) => label.type.id === 'customlabel')
```

Comment:
I'd put this up at the top of the function.  You could filter/warn/throw even before parsing the file if you got something unexpected